eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/core/Parameter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.core;
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public interface Parameter<T>
+{
+
+    void into(T container);
```

Comment:
Not sure what you mean by "dedicated interfaces for Intent and Bundle builder.", but you probably can use the same interface for both of them. The only thing that would be different is the name of the actual build method I guess.